File: server/datasets/java_source-code/Multiple-D1/organic_19.java
/*
 * Quick Sort - Organic Submission #19
 * Individual student implementation #19 with custom coding conventions.
 */

class Solution19 {
    public static int splitSegment19(int[] dataArray, int startBound, int endBound) {
        int pivotRef = dataArray[endBound];
        int partitionIdx = startBound - 1;
        for (int scanPtr = startBound; scanPtr < endBound; scanPtr++) {
            if (dataArray[scanPtr] <= pivotRef) {
                partitionIdx++;
                int hold = dataArray[partitionIdx];
                dataArray[partitionIdx] = dataArray[scanPtr];
                dataArray[scanPtr] = hold;
            }
        }
        int hold = dataArray[partitionIdx + 1];
        dataArray[partitionIdx + 1] = dataArray[endBound];
        dataArray[endBound] = hold;
        return partitionIdx + 1;
    }

    public static void performQuicksort19(int[] dataArray, int startBound, int endBound) {
        if (startBound < endBound) {
            int splitLoc = splitSegment19(dataArray, startBound, endBound);
            performQuicksort19(dataArray, startBound, splitLoc - 1);
            performQuicksort19(dataArray, splitLoc + 1, endBound);
        }
    }

    public static void main(String[] args) {
        int[] sample = {33, 7, 9, 14, 28, 42};
        performQuicksort19(sample, 0, sample.length - 1);
        System.out.println(java.util.Arrays.toString(sample));
    }
}

File: server/datasets/python_source-code/Multiple-D1/organic_19.py
"""
Quick Sort - Organic Submission #19
Individual student implementation #19 with custom coding conventions.
"""

def split_segment_19(data_array, start_bound, end_bound):
    pivot_reference = data_array[end_bound]
    partition_index = start_bound - 1
    for scan_pointer in range(start_bound, end_bound):
        if data_array[scan_pointer] <= pivot_reference:
            partition_index += 1
            data_array[partition_index], data_array[scan_pointer] = (
                data_array[scan_pointer],
                data_array[partition_index],
            )
    data_array[partition_index + 1], data_array[end_bound] = (
        data_array[end_bound],
        data_array[partition_index + 1],
    )
    return partition_index + 1

def perform_quicksort_19(data_array, start_bound, end_bound):
    if start_bound < end_bound:
        split_loc = split_segment_19(data_array, start_bound, end_bound)
        perform_quicksort_19(data_array, start_bound, split_loc - 1)
        perform_quicksort_19(data_array, split_loc + 1, end_bound)

if __name__ == "__main__":
    sample_data = [33, 7, 9, 14, 28, 42]
    perform_quicksort_19(sample_data, 0, len(sample_data) - 1)
    print("Sorted #19:", sample_data)
